contrib/kubespray/roles/requirement/infra/tasks/main.yml

```diff
@@ -0,0 +1,19 @@
+---
+- name: Output infra machines' os variables
+  debug:
+    var: "{{ item }}"
+  with_items:
+    - ansible_distribution
+    - ansible_distribution_version
+
+- name: Check ubuntu requirement. OpenPAI only supports ubuntu now.
+  include_tasks: ubuntu.yml
+  when: ansible_distribution == "Ubuntu" and ansible_distribution_version == "16.04"
+
+- name: Infra machines' os is not ubuntu.
+  fail:
+    msg: "Please ensure your infra machines' os is ubuntu 16.04 lts. 18.04 should be ok, but have not tested it yet"
```

Comment:
could you just output a warning instead of an error for ubuntu version > 16.04?
host os is not dependent as long as docker is installed, deployment only runs commands inside dev-box docker

Reply:
Done.